### PR TITLE
ReverseDiff support

### PR DESCRIPTION
**Pull request by mohamed82008**
This PR adds support for ReverseDiff.

### CODE REVIEW

**Comment by mohamed82008** [comment]
I believe this is ready for a review. Tests pass locally. Let's see what CI says.

File: src/filldist.jl [hunk 1]
```diff
@@ -11,7 +11,7 @@ const FillVectorOfUnivariate{
 function filldist(dist::UnivariateDistribution, N::Int)
     return product_distribution(Fill(dist, N))
 end
-filldist(d::Normal, N::Int) = MvNormal(fill(d.μ, N), d.σ)
+filldist(d::Normal, N::Int) = TuringMvNormal(fill(d.μ, N), d.σ)
```
Comment (devmotion) [inline src/filldist.jl]: Maybe use `Fill` instead of `fill`?

Comment (mohamed82008) [reply src/filldist.jl]: `Fill` doesn't work well with Tracker. `fill` gives a `TrackedArray`.

Comment (devmotion) [reply src/filldist.jl]: Ah OK. But isn't the point here that we don't need an actual array?

Comment (mohamed82008) [reply src/filldist.jl]: TrackedArrays are better for broadcasting. We have broadcasting in MvNormal logpdf.

File: src/reversediff.jl [hunk 2]
```diff
@@ -0,0 +1,235 @@
+const RTR = ReverseDiff.TrackedReal
+const RTV = ReverseDiff.TrackedVector
+const RTM = ReverseDiff.TrackedMatrix
+import SpecialFunctions: logbeta
+import Distributions: Gamma
+
+ReverseDiff.@forward binomlogpdf(n::Int, p::RTR, x::Int) = begin
```
Comment (yebai) [inline src/reversediff.jl]: Maybe add some comments explaining what's happening here.

Comment (mohamed82008) [reply src/reversediff.jl]: Sure, this is using ForwardDiff for the function.

File: src/filldist.jl (hunk 1 above)
Comment (devmotion) [reply src/filldist.jl]: I see, thanks for explaining!

**Pull request merged**